### PR TITLE
[Llama-3.3-70B Galaxy] Move all prefill trace creation to the beginning to avoid corruption in later batches

### PR DESCRIPTION
**Pull request by mtairum**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/30187

### Problem description
Seeing ND "bad" output when using repeat batches and batch_size = 32. This is showed by seeing output variability - as the same prompt sometimes leading to different outputs on different matches.

### What's changed
- Before, we were allocating a prefill trace after received the batched prompts. This meant if a new batch comes along and has a new prefill seqlen, we'd have to allocate a whole new trace, after running decode iterations for past batches.
  - This behaviour seems to be causing memory clobbering that could affect specific prompts and environments.
 
- This PR changes this behaviour and creates all available prefill traces (seqlens == [128, 1K, 2K, 4K]) at the beggining, i.e. during the first prefill call of the server.

### Checklist
- [ ] Galaxy APC: https://github.com/tenstorrent/tt-metal/actions/runs/18542423034
- [ ] Galaxy Choose your pipeline (PR-Model): https://github.com/tenstorrent/tt-metal/actions/runs/18537842601
- [ ] vLLM nightly CI run here: https://github.com/tenstorrent/tt-metal/actions/runs/18542245200 (other failures are irrelevant to this PR, WHGLX is passing)

### CODE REVIEW

File: models/demos/llama3_70b_galaxy/tt/generator.py [hunk 1]
```diff
@@ -108,7 +161,7 @@ def prefill_forward_text(
             and tt_out_logits_all_users is None
             and not return_logits
         ):
-            use_batched_prefill = True
+            use_batched_prefill = False
```
[the remark this answers is not in the document]
Comment (sraizada-tt) [reply models/demos/llama3_70b_galaxy/tt/generator.py]: are we turning this off?

File: models/demos/llama3_70b_galaxy/tt/generator.py [hunk 2]
```diff
@@ -118,7 +171,7 @@ def prefill_forward_text(
         for id, user_id in enumerate(all_users):
             logger.info(f"Prefilling User {user_id + 1}, use_batched_prefill: {use_batched_prefill}")
             if use_batched_prefill:
-                user_id = empty_slots
+                user_id = [user_id]
```
[the remark this answers is not in the document]
Comment (sraizada-tt) [reply models/demos/llama3_70b_galaxy/tt/generator.py]: why? I think this would break vllm continuous batching

**Review by sraizada-tt — APPROVED** [review]

File: .github/workflows/vllm-nightly-tests-impl.yaml [hunk 3]
```diff
@@ -93,7 +93,7 @@ jobs:
             runner-label: topology-6u,
             mesh-device: TG,
             tt-llama-text-ver: llama3_70b_galaxy,
-            override-tt-config: "{\"dispatch_core_axis\": \"col\", \"sample_on_device_mode\": \"all\", \"fabric_config\": \"FABRIC_1D_RING\", \"worker_l1_size\": 1344544, \"trace_region_size\": 95693824}",
+            override-tt-config: "{\"dispatch_core_axis\": \"col\", \"sample_on_device_mode\": \"all\", \"fabric_config\": \"FABRIC_1D_RING\", \"worker_l1_size\": 1344544, \"trace_region_size\": 184915840}",
```
Comment (skhorasganiTT) [inline .github/workflows/vllm-nightly-tests-impl.yaml:96]: Before merging, please update the vLLM command in the llama70b readme (https://github.com/tenstorrent/tt-metal/tree/main/models/demos/llama3_70b_galaxy#running-the-vllm-server) to reflect this change. And before or after merging, in a separate vLLM PR please update the value in the vLLM readme (at "The command to run Llama70B on Galaxy is:" under https://github.com/tenstorrent/vllm/tree/dev/tt_metal#llama-3132-1b-3b-8b-70b-and-qwen-25-7b-72b-text-models).

**Review by williamlyTT — APPROVED** [review]

**Review by skhorasganiTT — APPROVED** [review]

File: .github/workflows/vllm-nightly-tests-impl.yaml (hunk 3 above)
Comment (alingTT) [reply .github/workflows/vllm-nightly-tests-impl.yaml:96]: updated the tt-metal readme, will create seperate PR to update vLLM read me

**Comment by bgoelTT** [comment]
@tstescoTT this is the model-side prefill warmup we've been talking about. Just FYI we'll have some overlap for the first few trace captures.

~~Also @mtairum if this sweeps through all batch sizes, how much startup time does this add to the model?~~ Nevermind, saw that its (1, 32) not range()
